@slack/oauth fails when no bot scopes (only user scopes) are requested

### Description

Requesting only user scopes with `@slack/oauth` fails with stateSecret invalid even when it is valid.

```
[ERROR]  OAuth:InstallProvider:0 Error: The response from the authorization URL contained inconsistent information. Please file a bug.
    at InstallProvider.<anonymous> (MYPROJECT/node_modules/@slack/oauth/dist/index.js:333:27)
    at step (MYPROJECT/node_modules/@slack/oauth/dist/index.js:44:23)
    at Object.next (MYPROJECT/node_modules/@slack/oauth/dist/index.js:25:53)
    at fulfilled (MYPROJECT/node_modules/@slack/oauth/dist/index.js:16:58)
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
```

This fails (I tried everything here, even implementing my own state store before finding this bug)

```js
server.get('/', async (req, res) => {
    const url = await installer.generateInstallUrl({
        scopes: [ ],
        userScopes: ['channels:history', 'groups:history', 'mpim:history', 'im:history', 'chat:write'],
    })
    res.redirect(url)
})
```

while this succeeds

```js
server.get('/', async (req, res) => {
    const url = await installer.generateInstallUrl({
        scopes: ["chat:write"],
        userScopes: ['channels:history', 'groups:history', 'mpim:history', 'im:history', 'chat:write'],
    })
    res.redirect(url)
})
```

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Filling out the following details about bugs will help us solve your issue sooner.

#### Packages:

Select all that apply:

- [ ] `@slack/web-api`
- [ ] `@slack/events-api`
- [ ] `@slack/interactive-messages`
- [ ] `@slack/rtm-api`
- [ ] `@slack/webhooks`
- [x] `@slack/oauth`
- [ ] I don't know

#### Reproducible in:

package version: ^2.0.0

node version: v15.0.1

OS version(s): macOS 11.1

#### Steps to reproduce:

1. Write minimal `@slack/oauth` code and request only user scopes
2. See error

#### Expected result:

Authorisation succeeds

#### Actual result:

```
[ERROR]  OAuth:InstallProvider:0 Error: The response from the authorization URL contained inconsistent information. Please file a bug.
    at InstallProvider.<anonymous> (MYPROJECT/node_modules/@slack/oauth/dist/index.js:333:27)
    at step (MYPROJECT/node_modules/@slack/oauth/dist/index.js:44:23)
    at Object.next (MYPROJECT/node_modules/@slack/oauth/dist/index.js:25:53)
    at fulfilled (MYPROJECT/node_modules/@slack/oauth/dist/index.js:16:58)
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
```

--- a/packages/oauth/src/index.ts
+++ b/packages/oauth/src/index.ts
@@ -308,11 +308,13 @@ export class InstallProvider {
             v2Installation.enterpriseUrl = authResult.url;
           }
 
-        } else if (v2Resp.authed_user.access_token !== undefined && v2Resp.is_enterprise_install) {
-          // Only user scopes were requested & is an org enterprise install
+        } else if (v2Resp.authed_user.access_token !== undefined) {
+          // Only user scopes were requested
           // TODO: confirm if it is possible to do an org enterprise install without a bot user
           const authResult = await runAuthTest(v2Resp.authed_user.access_token, this.clientOptions);
-          v2Installation.enterpriseUrl = authResult.url;
+          if (v2Resp.is_enterprise_install) {
+            v2Installation.enterpriseUrl = authResult.url;
+          }
         } else {
           // TODO: make this a coded error
           throw new Error('The response from the authorization URL contained inconsistent information. Please file a bug.');

